File: python/algo/202508/algo2657.py
from typing import List

# 位运算 - 基础题
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        a, b = 0, 0
        ans = []
        for i, j in zip(A, B):
            a |= 1 << i
            b |= 1 << j
            ans.append((a&b).bit_count())
        return ans

File: cplus/leetcode/202508/algo2657.cpp
#include <bits/stdc++.h>
using namespace std;

// 位运算 - 基础题
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        long a = 0, b = 0; int n = A.size();
        vector<int> ans;
        for (int i=0; i<n; ++i){
            a |= (long)1 << A[i]; 
            b |= (long)1 << B[i];
            ans.push_back(__builtin_popcountl(a & b));
            // ans.push_back(popcount(a & b));
        }
        return ans;       
    }
};